Add settings for MongoDB server and some handling.

PredaaA opened the pull request:
This PR includes:
- `client` and `db` moved to cog init.
- Settings commands (`[p]levelerset` group) to choose a different host, port, credentials (username and password), and db name for MongoDB.
- That command group (`[p]levelerset`) show current setting, but only in DMs.
- Handling of wrong MongoDB host/port/credentials, instead of just error out it will let bot owner able use settings commands to set a new host/port. Meanwhile all other commands are blocked, but also all functions related to `self.db`.

aikaterna commented:
Commenting that there will be more to this PR so that's why I haven't looked at it yet.

aikaterna commented:
And it would have been great if I hit the right button...

PredaaA commented:
I just updated the PR with changes I added to Fixator repo, which includes:
- All commands of the `[p]levelerset` group are now using defaults. Like for port if you don't specify a port it'll set it to port 27017.
- Using `[p]levelerset` in DMs will show current settings.
- Add a setting to choose DB name (what's in `self.db`).
- Changed `_ready_event` to `_db_ready`, to use simply a bool instead of `asyncio.Event()`, then return in all functions using db, if db isn't ready instead of using `wait` of `asyncio.Event()`.
- Also if db isn't ready, `cog_before_invoke` got replaced to `cog_check`, and still with `[p]levelerset` group available, meanwhile all other commands are not usable.
